src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -472,6 +487,12 @@ def generate_config(
         }
 
         if self.mgr.secure_monitoring_stack:
+            r2: Dict[str, Any] = {'files': {}}
+            for url, credentials in clusters_credentials.items():
+                r2['files'][f'prometheus_{url}_cert.crt'] = credentials['certificate']
+                credentials['cert_file_name'] = f'prometheus_{url}_cert.crt'
```

Comment:
Please, keep in mind that building the filename (not sure how it would be used later) based directly on the url could lead to issues with the generated filename in case the url has some special characters.

Reply:
Updated the PR